crates/re_build_info/src/crate_version.rs

```diff
@@ -1,94 +1,191 @@
-/// We disallow version numbers larger than this in order to keep a few bits for future use.
-///
-/// If you are running up against this limit then feel free to bump it!
-const MAX_NUM: u8 = 31;
+mod meta {
+    pub const TAG_MASK: u8 = 0b11000000;
+    pub const VALUE_MASK: u8 = 0b00011111;
 
-const IS_ALPHA_BIT: u8 = 1 << 7;
-const IS_PRERELEASE_BIT: u8 = 1 << 6;
+    pub const RC: u8 = 0b01000000;
+    pub const ALPHA: u8 = 0b10000000;
+    pub const DEV_ALPHA: u8 = 0b11000000;
+}
 
 /// The version of a Rerun crate.
 ///
-/// Sub-set of semver supporting `major.minor.patch` plus an optional `-alpha.X`.
-///
-/// When parsing, any `+metadata` suffix is ignored.
+/// Sub-set of semver supporting `major.minor.patch-{alpha,rc}.N+dev`.
 ///
-/// Examples: `1.2.3`, `1.2.3-alpha.4`.
+/// The string value of build metadata is not preserved.
 ///
-/// We use `-alpha.X` when we publish pre-releases to crates.io and PyPI.
+/// Examples: `1.2.3`, `1.2.3-alpha.4`, `1.2.3-alpha.1+dev`.
 ///
-/// We use a `+githash` suffix for continuous pre-releases that you can download from our GitHub.
-/// We do NOT store that in this struct. See also `scripts/ci/version_util.py`.
+/// `-alpha.N+dev` versions are used for local or CI builds.
+/// `-alpha.N` versions are used for weekly releases.
+/// `-rc.N` versions are used for release candidates as we're preparing for a full release.
 ///
-/// The version numbers aren't allowed to be very large (current max: 31).
-/// This limited subset it chosen so that we can encode the version in 32 bits
+/// The version numbers (`N`) aren't allowed to be very large (current max: 31).
+/// This limited subset is chosen so that we can encode the version in 32 bits
 /// in our `.rrd` files and on the wire.
+///
+/// Here is the current binary format:
+/// ```text,ignore
+/// major    minor    patch    meta
+/// 00000000 00000000 00000000 00000000
+///                            ▲▲ ▲   ▲
+///                            ││ └┬──┘
+///                            ││  └─ N
+///                            │└─ rc/dev
+///                            └─ alpha
+/// ```
+///
+/// The valid bit patterns for `meta` are:
+/// - `100NNNNN` -> `-alpha.N`
+/// - `110NNNNN` -> `-alpha.N+dev`
+/// - `010NNNNN` -> `-rc.N`
+/// - `00000000` -> none of the above
```

Comment:
why is the third most significant bit left zero in all cases, overflow check?

Reply:
Free real estate. But I guess it doesn't matter because the format is still considered unstable so we're free to keep breaking it as we need to. I doubt we'd hit the limit of 31 rc builds, but I'm still going to bump this to use that extra bit and increase the limit to 63.